server/server_test.go

```diff
@@ -723,36 +740,72 @@ func (f *FakeLogger) assertNotContains(t *testing.T, content string) {
 func TestLogSourceIPs(t *testing.T) {
 	var level log.Level
 	require.NoError(t, level.Set("info"))
-	logger := newFakeLogger()
 	cfg := Config{
 		HTTPMiddleware:   []middleware.Interface{middleware.Log{Log: log.Global()}},
 		MetricsNamespace: "testing_mux",
 		LogLevel:         level,
-		Log:              logger,
 		LogSourceIPs:     true,
 	}
 	setAutoAssignedPorts(DefaultNetwork, &cfg)
 
-	server, err := New(cfg)
-	require.NoError(t, err)
+	startServer := func(cfg Config) *Server {
+		prometheus.DefaultRegisterer = prometheus.NewRegistry()
+		server, err := New(cfg)
+		require.NoError(t, err)
 
-	server.HTTP.HandleFunc("/error500", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(500)
+		server.HTTP.HandleFunc("/error500", func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(500)
+		})
+
+		go func() {
+			require.NoError(t, server.Run())
+		}()
+
+		return server
+	}
+
+	t.Run("without PROXY protocol", func(t *testing.T) {
+		logger := newFakeLogger()
+		cfg.Log = logger
+
+		server := startServer(cfg)
+		defer server.Shutdown()
+
+		logger.assertNotContains(t, "sourceIPs")
+
+		req, err := http.NewRequest("GET", httpTarget(server, "/error500"), nil)
+		require.NoError(t, err)
+		_, err = http.DefaultClient.Do(req)
+		require.NoError(t, err)
+
+		logger.assertContains(t, "sourceIPs=127.0.0.1")
 	})
 
-	go func() {
-		require.NoError(t, server.Run())
-	}()
-	defer server.Shutdown()
+	t.Run("without PROXY protocol", func(t *testing.T) {
```

Comment:
The name of the test is wrong: should be `with PROXY protocol` (not without) as per line 787.